Mathlib/Algebra/Algebra/Quasispectrum.lean

```diff
@@ -268,9 +268,9 @@ where `R` is a *semi*ring, but `φ` must still function over a scalar ring `S`.
 need `S` to be explicit. The primary use case is, for instance, `R := ℝ≥0` and `S := ℝ` or
 `S := ℂ`. -/
 lemma NonUnitalAlgHom.quasispectrum_apply_subset' {F R : Type*} (S : Type*) {A B : Type*}
-    [CommSemiring R] [Semiring S] [NonUnitalRing A] [NonUnitalRing B] [Module R S]
-    [Module S A] [Module R A] [Module S B] [Module R B] [IsScalarTower R S A] [IsScalarTower R S B]
-    [FunLike F A B] [NonUnitalAlgHomClass F S A B] (φ : F) (a : A) :
+    [CommSemiring R] [Monoid S] [NonUnitalRing A] [NonUnitalRing B] [SMul R S]
+    [DistribMulAction S A] [Module R A] [DistribMulAction S B] [Module R B] [IsScalarTower R S A]
+    [IsScalarTower R S B] [FunLike F A B] [NonUnitalAlgHomClass F S A B] (φ : F) (a : A) :
```

Comment:
This one feels a little vacuous to me, I can't think of any semirings that act on a monoid. This is more in @j-loreaux's area of expertise though.

Reply:
For simplicity I just reverted the changes in this file.